data/StateMachine.j2119

```diff
@@ -166,10 +166,18 @@ A Wait State MAY have a referencePath field named "TimestampPath".
 A Wait State MUST have only one of "Seconds", "SecondsPath", "Timestamp", and "TimestampPath".
 A Wait State MUST have a field named one of "Seconds", "SecondsPath", "Timestamp", or "TimestampPath".
 A Parallel State MUST have an object-array field named "Branches"; each element is a "Branch".
-A Map State MUST have an object field named "Iterator"; its value is a "Branch".
+A Map State MAY have an object field named "Iterator"; its value is a "Branch".
```

Comment:
Iterator should not be type Branch, it should also be type "Processor". You could, for example, have `ProcessorConfig` in `Iterator`, which `Branch` does not have.

While you're here, maybe rename "Processor" to "ItemProcessor", for some extra precision in naming?

Reply:
might be an idea to add test coverage for this also.

Reply:
Not sure if I follow your first suggestion - the existing code says that an Iterator is type Branch. The change I made here is `must` -> `may` to account for Maps that have ItemProcessor instead of Iterator.

Reply:
yes, I know it's existing code... but there's a pre-existing bug here. 

Sorry, I didn't mean to imply you _caused_ it :-) Just that here's an opportunity to fix it, if you're game :-) 

The deprecated "Iterator" field does allow the `ProcessorConfig` field (which `Branch` does not), so it's more properly also an `ItemProcessor`.